lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class ReplicaSet < KubernetesResource
+    TIMEOUT = 5.minutes
+
+    def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
+      @name = name
+      @namespace = namespace
+      @context = context
+      @file = file
+      @parent = parent
+      @logger = logger
+      @pods = []
+      @rollout_data = { "replicas" => 0 }
+    end
+
+    def sync
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
+
+      if @found = st.success?
+        rs_data = JSON.parse(raw_json)
+        interpret_json_data(rs_data)
+      else # reset
+        @rollout_data = { "replicas" => 0 }
+        @status = nil
+        @pods = []
+      end
+    end
+
+    def interpret_json_data(rs_data)
+      pods_data = get_pods(rs_data)
+
+      pods_data.each do |pod_data|
+        pod = Pod.new(
+          name: pod_data["metadata"]["name"],
+          namespace: namespace,
+          context: context,
+          file: nil,
+          parent: "#{@name.capitalize} replica set",
+          logger: @logger
+        )
+        pod.deploy_started = @deploy_started
+        pod.interpret_json_data(pod_data)
+        @pods << pod
+      end
+
+      @rollout_data.merge!(rs_data["status"]
+        .slice("replicas", "availableReplicas", "readyReplicas"))
+      @status = @rollout_data.map { |st, num| "#{num} #{st.chop.pluralize(num)}" }.join(", ")
+    end
+
+    def deploy_succeeded?
+      @rollout_data["replicas"].to_i == @rollout_data["availableReplicas"].to_i &&
+      @rollout_data["replicas"].to_i == @rollout_data["readyReplicas"].to_i
+    end
+
+    def deploy_failed?
+      @pods.present? && @pods.all?(&:deploy_failed?)
+    end
+
+    def deploy_timed_out?
+      super || @pods.present? && @pods.all?(&:deploy_timed_out?)
+    end
+
+    def exists?
+      unmanaged? ? @found : true
+    end
+
+    def fetch_events
+      own_events = super
+      return own_events unless @pods.present?
+      own_events.merge(@pods.first.fetch_events)
+    end
+
+    def fetch_logs
+      return {} unless container_names.present?
+      container_names.each_with_object({}) do |container_name, container_logs|
+        out, _err, _st = kubectl.run(
+          "logs",
+          id,
+          "--container=#{container_name}",
+          "--since-time=#{@deploy_started.to_datetime.rfc3339}",
+          "--tail=#{LOG_LINE_COUNT}"
+        )
+        container_logs["#{id}/#{container_name}"] = out
```

Comment:
How about this?

![image](https://user-images.githubusercontent.com/4789493/27105516-da377eba-505d-11e7-8629-73ac201192b6.png)

- each gets its own section, with a heading instead of tags on every line
- sorted by line count
- identifier format `container 'foo'` instead of the full resource path (`Deployment/web/foo`)